File: fix_apex.py
import os
import urllib.request
import json
import ssl

def get_env():
    env = {}
    try:
        with open('.env', 'r') as f:
            for line in f:
                if '=' in line and not line.startswith('#'):
                    k, v = line.strip().split('=', 1)
                    k = k.strip()
                    v = v.strip().strip('"').strip("'")
                    env[k] = v
    except Exception as e:
        print(f"Error reading .env: {e}")
    return env

env = get_env()
API_KEY = env.get('PORKBUN_API_KEY')
SECRET_KEY = env.get('PORKBUN_SECRET_KEY')
DOMAIN = 'chris.quest'

def request(path, payload={}):
    data = {"apikey": API_KEY, "secretapikey": SECRET_KEY}
    data.update(payload)
    data = json.dumps(data).encode('utf-8')
    
    url = f"https://api.porkbun.com/api/json/v3/dns/{path}"
    req = urllib.request.Request(url, data=data, headers={'Content-Type': 'application/json'})
    
    ctx = ssl.create_default_context()
    ctx.check_hostname = False
    ctx.verify_mode = ssl.CERT_NONE
    
    try:
        with urllib.request.urlopen(req, context=ctx) as response:
            return json.loads(response.read().decode())
    except urllib.error.HTTPError as e:
        print(f"HTTP Error {e.code}: {e.reason} -> {e.read().decode()}")
        return None
    except Exception as e:
        print(f"Request failed: {url} - {e}")
        return None

def fix_records():
    print(f"Fixing DNS for {DOMAIN}... (API_KEY: {API_KEY[:5]}...)")
    
    records_response = request(f"retrieve/{DOMAIN}")
    if not records_response or records_response.get('status') != 'SUCCESS':
        print("Failed to pull records.")
        return

    records = records_response.get('records', [])
    
    # 1. Delete Porkbun Parked IPs
    for r in records:
        if r['name'] == DOMAIN and r['type'] == 'A' and r['content'] == '119.8.104.22':
            print(f"Deleting parked IP {r['id']}...")
            request(f"delete/{DOMAIN}/{r['id']}")

    # 2. Add Netlify Load Balancer (75.2.60.5)
    netlify_a = any(r['name'] == DOMAIN and r['type'] == 'A' and r['content'] == '75.2.60.5' for r in records)
    if not netlify_a:
        print("Adding Netlify Apex A Record (75.2.60.5)...")
        res = request(f"create/{DOMAIN}", {"name": "", "type": "A", "content": "75.2.60.5", "ttl": "600"})
        print(res)
    else:
        print("Netlify Apex A Record already exists.")

    print("✅ DNS checks completed.")

if __name__ == '__main__':
    fix_records()

File: fix_apex.js
const fs = require('fs');
const https = require('https');

// Simple .env parser to avoid dot-env dependency
function getEnv() {
    const file = fs.readFileSync('.env', 'utf-8');
    const env = {};
    file.split('\n').forEach(line => {
        const [key, ...rest] = line.split('=');
        if (key && rest.length > 0) {
            env[key.trim()] = rest.join('=').trim();
        }
    });
    return env;
}

const env = getEnv();
const API_KEY = env.PORKBUN_API_KEY;
const SECRET_KEY = env.PORKBUN_SECRET_KEY;
const DOMAIN = 'chris.quest';

function request(path, payload = {}) {
    return new Promise((resolve, reject) => {
        const data = JSON.stringify({
            apikey: API_KEY,
            secretapikey: SECRET_KEY,
            ...payload
        });

        const options = {
            hostname: 'api.porkbun.com',
            port: 443,
            path: '/api/json/v3/domain/dns/' + path,
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
                'Content-Length': Buffer.byteLength(data)
            }
        };

        const req = https.request(options, (res) => {
            let result = '';
            res.on('data', (d) => { result += d; });
            res.on('end', () => {
                resolve(JSON.parse(result));
            });
        });

        req.on('error', reject);
        req.write(data);
        req.end();
    });
}

async function fixRecords() {
    console.log(`Fixing DNS for ${DOMAIN}...`);
    try {
        const records = await request(`retrieve/${DOMAIN}`);
        if (records.status !== 'SUCCESS') {
            console.error('Failed to retrieve records:', records);
            return;
        }

        const parkingRecords = records.records.filter(r => r.name === DOMAIN && r.type === 'A' && r.content === '119.8.104.22');
        for (const r of parkingRecords) {
            console.log(`Deleting parked IP record ${r.id}...`);
            await request(`delete/${DOMAIN}/${r.id}`);
        }

        // Check if Nelify ALIAS A is present
        const netlifyA_Record = records.records.find(r => r.name === DOMAIN && r.type === 'A' && r.content === '75.2.60.5');
        if (!netlifyA_Record) {
            console.log(`Adding Netlify ALIAS/A record...`);
            const addResult = await request(`create/${DOMAIN}`, {
                name: '',
                type: 'A',
                content: '75.2.60.5',
                ttl: 600
            });
            console.log('Result:', addResult);
        } else {
            console.log('Network ALIAS/A record already exists.');
        }

        // Also check wildcard and www just in case
        const wwwRecord = records.records.find(r => r.name === `www.${DOMAIN}`);
        if (!wwwRecord) {
            console.log(`Adding Netlify WWW CNAME...`);
            await request(`create/${DOMAIN}`, {
                name: 'www',
                type: 'CNAME',
                content: 'dazzling-semifreddo.netlify.app',
                ttl: 600
            });
        } else {
            console.log('WWW record already exists.');
        }

        console.log('✅ DNS records fixed. Propagation takes about 5 minutes.');
    } catch (err) {
        console.error('Error:', err);
    }
}

fixRecords();
